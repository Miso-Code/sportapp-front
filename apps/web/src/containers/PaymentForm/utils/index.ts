export function generateLabelValuePairs(n: number) {
	const pairs = []
	for (let i = 1; i <= n; i++) {
		pairs.push({ label: i.toString(), value: i.toString() })
	}
	return pairs
}
