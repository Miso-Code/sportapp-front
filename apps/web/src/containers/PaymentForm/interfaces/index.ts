import { FormPaymentData } from '../utils/schema'

export interface Props {
	onSubmit: (data: FormPaymentData) => void
	onCancel: () => void
	className?: string
	options: {
		label: string
		value: string
	}[]
	price?: number
	isLoading?: boolean
}
