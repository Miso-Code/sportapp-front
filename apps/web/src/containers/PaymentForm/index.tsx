import SelectController from '@/components/Inputs/SelectController'
import TextFieldController from '@/components/Inputs/TexFieldController'
import { yupResolver } from '@hookform/resolvers/yup'
import LoadingButton from '@mui/lab/LoadingButton'
import { useState } from 'react'
import Cards, { Focused } from 'react-credit-cards-2'
import 'react-credit-cards-2/dist/lib/styles.scss'
import { useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import { generateLabelValuePairs } from './utils'
import { FormPaymentData, paymentSchema } from './utils/schema'
import { Props } from './interfaces'

const amountOptions = generateLabelValuePairs(10)

export default function PaymentForm({
	onCancel,
	onSubmit: onSubmitProp,
	...props
}: Props) {
	const { t } = useTranslation()
	const [focus, setFocus] = useState<Focused>('name')
	const { watch, handleSubmit, control } = useForm({
		resolver: yupResolver(paymentSchema),
		mode: 'onChange'
	})

	const cardNumber = watch('number', '')
	const cardHolder = watch('name', '')
	const expirationDate = watch('expiry', '')
	const cvv = watch('cvc', '')

	const onSubmit = (data: FormPaymentData) => {
		onSubmitProp(data)
	}

	const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
		setFocus(e.target.name as Focused)
	}

	return (
		<form onSubmit={handleSubmit(onSubmit)} {...props}>
			<Cards
				cvc={cvv}
				expiry={expirationDate}
				name={cardHolder}
				number={cardNumber}
				placeholders={{ name: t('form.card.cardHolderPlaceholder') }}
				focused={focus}
			/>
			<section className='flex flex-col gap-4 p-4 mt-2'>
				<SelectController
					control={control}
					isTranslated={false}
					selectProps={{ fullWidth: true }}
					label={t('form.amount')}
					name='amount'
					options={amountOptions}
				/>
				<TextFieldController
					control={control}
					fullWidth
					label={t('form.card.cardNumber')}
					type='number'
					method='maxLength'
					maxLength={16}
					onFocus={handleFocus}
					name='number'
				/>
				<TextFieldController
					control={control}
					fullWidth
					label={t('form.card.cardHolder')}
					type='text'
					onFocus={handleFocus}
					name='name'
				/>
				<div className='flex gap-4'>
					<TextFieldController
						control={control}
						className='w-1/2'
						method='expirationDate'
						onFocus={handleFocus}
						label={t('form.card.expirationDate')}
						name='expiry'
					/>

					<TextFieldController
						control={control}
						className='w-1/2'
						method='cvv'
						onFocus={handleFocus}
						label={t('form.card.cvv')}
						name='cvc'
					/>
				</div>
				<div className='flex md:flex-row gap-4 mt-2'>
					<LoadingButton
						fullWidth
						size='large'
						color='error'
						type='button'
						onClick={onCancel}
						variant='contained'>
						{t('form.cancel')}
					</LoadingButton>
					<LoadingButton
						fullWidth
						size='large'
						color='success'
						type='submit'
						variant='contained'>
						{t('form.checkout')}
					</LoadingButton>
				</div>
			</section>
		</form>
	)
}
