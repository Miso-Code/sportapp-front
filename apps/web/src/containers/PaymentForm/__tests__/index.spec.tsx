import PaymentForm from '..'
import {
	fireEvent,
	render,
	RenderResult,
	waitFor
} from '@testing-library/react'

describe('PaymentForm', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<PaymentForm onCancel={jest.fn()} onSubmit={jest.fn()} />
		)
	})

	it('should render component', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should call cancel function', async () => {
		const handleCancel = jest.fn()
		wrapper.rerender(
			<PaymentForm onCancel={handleCancel} onSubmit={jest.fn()} />
		)
		const cancelButton = wrapper.getByText('form.cancel')
		fireEvent.click(cancelButton)
		await waitFor(() => expect(handleCancel).toHaveBeenCalled())
	})

	it('should show card front and focus un cardNumber', async () => {
		const cardNumber = wrapper.container.querySelector(
			'input[name="number"]'
		)
		fireEvent.focus(cardNumber as Element)
		await waitFor(() => {
			const cardFrontNumber = wrapper.container.querySelector('.rccs__number')

			expect(cardFrontNumber).toHaveClass('rccs--focused')
		})
	})
})
