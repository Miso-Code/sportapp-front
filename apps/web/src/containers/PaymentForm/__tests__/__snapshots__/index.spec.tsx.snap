// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentForm should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="rccs"
        >
          <div
            class="rccs__card rccs__card--unknown"
          >
            <div
              class="rccs__card--front"
            >
              <div
                class="rccs__card__background"
              />
              <div
                class="rccs__issuer"
              />
              <div
                class="rccs__cvc__front"
              />
              <div
                class="rccs__number"
              >
                •••• •••• •••• ••••
              </div>
              <div
                class="rccs__name rccs--focused"
              >
                form.card.cardHolderPlaceholder
              </div>
              <div
                class="rccs__expiry"
              >
                <div
                  class="rccs__expiry__valid"
                >
                  valid thru
                </div>
                <div
                  class="rccs__expiry__value"
                >
                  ••/••
                </div>
              </div>
              <div
                class="rccs__chip"
              />
            </div>
            <div
              class="rccs__card--back"
            >
              <div
                class="rccs__card__background"
              />
              <div
                class="rccs__stripe"
              />
              <div
                class="rccs__signature"
              />
              <div
                class="rccs__cvc"
              />
              <div
                class="rccs__issuer"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1oke1pi"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="amount-select-label"
            >
              form.amount
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-controls=":r0:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="amount-select-label mui-component-select-amount"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-amount"
                role="combobox"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                id="amount"
                name="amount"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    form.amount
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            />
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              form.card.cardNumber
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="number"
                name="number"
                type="number"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    form.card.cardNumber
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              form.card.cardHolder
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="name"
                name="name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    form.card.cardHolder
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-p27s4m"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r3:"
                id=":r3:-label"
              >
                form.card.expirationDate
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="expiry"
                  name="expiry"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      form.card.expirationDate
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r4:"
                id=":r4:-label"
              >
                form.card.cvv
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="cvc"
                  name="cvc"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      form.card.cvv
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-ke9lij"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorError MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorError MuiButton-fullWidth css-1e0debn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              id=":r5:"
              tabindex="0"
              type="button"
            >
              form.cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSuccess MuiButton-fullWidth Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSuccess MuiButton-fullWidth css-xtcf48-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              disabled=""
              id=":r6:"
              tabindex="-1"
              type="submit"
            >
              form.checkout
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="rccs"
      >
        <div
          class="rccs__card rccs__card--unknown"
        >
          <div
            class="rccs__card--front"
          >
            <div
              class="rccs__card__background"
            />
            <div
              class="rccs__issuer"
            />
            <div
              class="rccs__cvc__front"
            />
            <div
              class="rccs__number"
            >
              •••• •••• •••• ••••
            </div>
            <div
              class="rccs__name rccs--focused"
            >
              form.card.cardHolderPlaceholder
            </div>
            <div
              class="rccs__expiry"
            >
              <div
                class="rccs__expiry__valid"
              >
                valid thru
              </div>
              <div
                class="rccs__expiry__value"
              >
                ••/••
              </div>
            </div>
            <div
              class="rccs__chip"
            />
          </div>
          <div
            class="rccs__card--back"
          >
            <div
              class="rccs__card__background"
            />
            <div
              class="rccs__stripe"
            />
            <div
              class="rccs__signature"
            />
            <div
              class="rccs__cvc"
            />
            <div
              class="rccs__issuer"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1oke1pi"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="amount-select-label"
          >
            form.amount
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-controls=":r0:"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="amount-select-label mui-component-select-amount"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-component-select-amount"
              role="combobox"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              id="amount"
              name="amount"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  form.amount
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            form.card.cardNumber
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="number"
              name="number"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  form.card.cardNumber
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r2:"
            id=":r2:-label"
          >
            form.card.cardHolder
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="name"
              name="name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  form.card.cardHolder
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-p27s4m"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r3:"
              id=":r3:-label"
            >
              form.card.expirationDate
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="expiry"
                name="expiry"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    form.card.expirationDate
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r4:"
              id=":r4:-label"
            >
              form.card.cvv
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="cvc"
                name="cvc"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    form.card.cvv
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-ke9lij"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorError MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorError MuiButton-fullWidth css-1e0debn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            id=":r5:"
            tabindex="0"
            type="button"
          >
            form.cancel
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSuccess MuiButton-fullWidth Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSuccess MuiButton-fullWidth css-xtcf48-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            disabled=""
            id=":r6:"
            tabindex="-1"
            type="submit"
          >
            form.checkout
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
